### PR TITLE
EVG-16479 Update merge logic for build variants

### PR DESCRIPTION
**Pull request by bynn**
[EVG-16479](https://jira.mongodb.org/browse/EVG-16479)

### Description 
allow users to add tasks to bv before defining it in one of the include files

### Testing 
unit test

### CODE REVIEW

File: model/project_parser_merge_functions.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (pp *ParserProject) mergeBuildVariant(toMerge *ParserProject) error {
 	}
 	for _, bv := range toMerge.BuildVariants {
 		if currentBV, ok := bvs[bv.Name]; ok {
-			if !bv.canMerge() {
+			if !currentBV.canMerge() && !bv.canMerge() {
 				catcher.Errorf("build variant '%s' has been declared already", bv.Name)
```
[the remark this answers is not in the document]
Comment (bynn) [reply model/project_parser_merge_functions.go]: done

File: model/project_parser.go [hunk 2]
```diff
@@ -393,7 +393,6 @@ func (pbv *parserBV) canMerge() bool {
 		pbv.CronBatchTime == "" &&
 		pbv.Stepback == nil &&
 		pbv.RunOn == nil &&
-		pbv.DisplayTasks == nil &&
```
[the remark this answers is not in the document]
Comment (bynn) [reply model/project_parser.go:396]: we would only want to not merge the build variants if only one has the non-task declared and i made a mistake when i first wrote this of checking if display tasks were not defined as well

[the remark this answers is not in the document]
Comment (ablack12) [reply model/project_parser.go:396]: So should line 385 be more like (len(pbv.Task) +len(pbv.DisplayTasks) > 0 ?

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/project_parser.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bynn) [reply model/project_parser.go:396]: yeah i think that makes sense but i kept it an != 0 because i thought it was more clear.

**Review by ablack12 — APPROVED** [review]

**Pull request merged**